### ISSUE TITLE
SDN - KLAB2 and EMERALD clusters cannot access apps.smtp.gov.bc.ca for sending emails

### ISSUE THREAD

**Issue by wmhutchison** [BCDevOps/developer-experience#3105]
**Describe the issue**
Once the initial dust had cleared for these new NSXT-based clusters in Openshift AlertManager, it was discovered that AlertManager could not send emails through the standard email relay apps.smtp.gov.bc.ca. Investigation revealed that regular firewalls are dropping these email attempts, likely by virtue that internal IPs are now attempting to connect to these resources where normally no such thing would happen. Will need to confirm this is the issue and not NSX-T blockage, and open a suitable ticket tor request firewall changes.

**Additional context**
DXC Tickets opened to start this request: RITM0127845/SCTASK0267920
MySC tickets opened by DXC Service Desk to pass the firewall request along to OCIO Messaging: SCTASK0468470

**Definition of done**
- [x] Confirm this is a legacy firewall issue and not one involving NSX guard-rails or DFW blocks.
- [x] Open ticket with DXC Service Desk to request from all pod and node IP ranges for both KLAB2 and EMERALD.
- [x] Follow up with DSC Service Desk to request status update on request.
- [x] Engage OCIO Messaging regarding concerns involving the access.
- [x] Provide OCIO Messaging with requested Firewall Object Names for SDN source IPs.
- [x] Wait for OCIO Messaging to file and fulfill firewall request to implement requested access.
- [x] Work with Messaging as to why manual SMTP tests now work but AlertManager cannot send emails.
- [x] Test that AlertManager on both KLAB2 and EMERALD can now send emails to gov.bc.ca.
- [x] Open internal INC with DXC Service Desk to determine why we are not receiving AlertManager emails.

**Comment by wmhutchison** [comment]
Received a follow-up from Chris Blackhall on the OCIO Messaging team regarding the firewall access request. As probably should be suspected, this team has no knowledge of the SDN project thus is asking exactly why they should be allowing IP ranges normally not considered routable and usable in a data centre to connect to their email services. 

Have included Thom Bagnall on the responses to date so that he might possibly cut some of the red tape on the validity of this request. Will follow up again today since no new responses have been received yet on this matter.

**Comment by wmhutchison** [comment]
Re-assigning to Steven to follow up on this while I am off on vacation.

**Comment by wmhutchison** [comment]
Moving back to Blocked and will update shortly with its status and add/update a suitable "Blocked By" to the ticket.

**Comment by wmhutchison** [comment]
Last known update for this is that Thom Bagnall and Brian Price are going to meet with the OCIO Messaging group on October 25th to discuss this issue further. DXC has provided all the technical details possible on the matter.

**Comment by wmhutchison** [comment]
Unblocked on addressing OCIO Messaging concerns about allowing the source IP access to their SMTP services. Remaining blocked while we wait for them to put in the firewall request to allow the access. Will unblock once fulfilled and we're able to test the access.

**Comment by wmhutchison** [comment]
firewall request now fulfilled. Confirmed via manual netcat test to simulate an SMTP conversation that we indeed are now able technically to send emails from a private IP via apps.smtp.gov.bc.ca, yet we have yet to receive any alerts from AlertManager.

Updated the email thread with OCIO Messaging asking them to review their mail server logs and giving them the source IP ranges for KLAB2/EMERALD network spaces for the openshift-monitoring namespace, so they can review logs and determine if something else is not allowing the AlertManager emails from getting through, and if so, if it's something we can remediate from our end or not.

**Comment by wmhutchison** [comment]
Received a response from Messaging requesting the From address and example of what a Subject line would look like. Not asking for source IP tells me indirectly that the edge servers responsible are accepting the emails, but for some reason, the emails are not being allowed through. Only the edge servers care about the source IP address, after that it's a function of content or other policies which may or may not be allowing content through.

**Comment by wmhutchison** [comment]
Earlier this week, followed up with OCIO Messaging with the output of a manual SMTP conversation with their email services proving that emails are being accepted. Reverted to using a From address involving @dxcas.com as the original From address had too many hits in their searches.

Will push for escalation via OCIO stakeholders later this week if no response or resolution is achieved soon.

Also am now including AdvSol Service Desk email address in all correspondence so they no longer need to ask me separately for updates.

**Comment by wmhutchison** [comment]
No response received to date on this matter from OCIO Messaging. Dan Deane has this item on his list of things to follow up on today in a separate Governance meeting.

**Comment by wmhutchison** [comment]
Heard back from OCIO Messaging. A test email sent on November 14th via telnet from an EMERALD pod via apps.smtp.gov.bc.ca was according to their logs, received and delivered successfully to its destination (whatever is handling emails for dxcas.com).

Asked Cailey Jones to volunteer as a test subject with a gov.bc.ca address and re-tested the manually-created email originally sent on November 14th. She confirmed receipt of the email in her Outlook, though it was labeled as spam.

This has narrowed this issue down to be specific to DXC and AlertManager. While it did start out as an issue with firewall access, that has been since fixed.

A new ticket within DXC has been opened (INC0073249) and if all parties are agreeable, will close off the current support tickets along with this current ZenHub ticket, and replace with a new ZenHub ticket that has a more specific scope.

Cailey's assistance for testing did affirm that EMERALD has no general issues sending out emails, crossing that off of validation tests. :)

**Comment by wmhutchison** [comment]
Issue now resolved. While it is felt that adjusting the From addresses may have helped (used to be hsmcs@dxcas.com, is now PlatformServicesTeam@gov.bc.ca), the big fix was modifying AlertManager config to remove

```
matchers: ['openshift_io_alert_source!="platform"']
```
and replace with
```
match:
  prometheus: openshift-user-workload-monitoring/user-workload
```

The former it appears works with AlertManager on Openshift 4.10 but not with Openshift 4.9 which is what KLAB2 and EMERALD are still running. Unfortunately AlertManager logs gave zero clues  to this issue, we came across this specific fix by accident when re-applying an ansible playbook whose remote branch contents didn't have the newest updates, thus resolving KLAB2 and pointing us in the correct direction.

**Comment by wmhutchison** [comment]
Issue now resolved. While it is felt that adjusting the From addresses may have helped (used to be hsmcs@dxcas.com, is now PlatformServicesTeam@gov.bc.ca), the big fix was modifying AlertManager config to remove

```
matchers: ['openshift_io_alert_source!="platform"']
```
and replace with
```
match:
  prometheus: openshift-user-workload-monitoring/user-workload
```

The former it appears works with AlertManager on Openshift 4.10 but not with Openshift 4.9 which is what KLAB2 and EMERALD are still running. Unfortunately AlertManager logs gave zero clues  to this issue, we came across this specific fix by accident when re-applying an ansible playbook whose remote branch contents didn't have the newest updates, thus resolving KLAB2 and pointing us in the correct direction.
